builtins/amp-img.js

```diff
@@ -64,17 +64,12 @@ export class AmpImg extends BaseElement {
   }
 
   /** @override */
-  prerenderAllowed() {
-    return true;
-  }
-
-  /** @override */
   isRelayoutNeeded() {
     return true;
   }
 
   /** @override */
-  layoutCallback() {
+  prerenderCallback() {
```

Comment:
To start with, we should confirm two things:

1. We need to make sure that "prerendered" elements are not marked as "laid out". In the current implementation they are.
2. We need to decide whether we will always run "prerender". Obviously we will run it if we are NOT ready to layout. But we will we run it along with layout as well? It looks like so far we DO want to run it always because layout can take a lot longer.

Also, some approaches such as `placeholder` can yield better results since runtime will manage them better w/o assistance. For instance, in this PR `amp-youtube` is kind of regressing - currently we add placeholder in the `buildCallback` and it immediately becomes the responsibility of the runtime and its rules. Let's review the effect on other elements one more time to make sure we are putting the right intentions into the prerender callback.